examples/erc20Dividend.ts

```diff
@@ -117,53 +105,99 @@ window.addEventListener('load', async () => {
     name: ModuleName.ERC20DividendCheckpoint,
     address: erc20DividendAddress,
   });
-  const erc20Dividend = (await tickerSecurityTokenInstance.getModulesByName({
-    moduleName: ModuleName.ERC20DividendCheckpoint,
+
+  // Get General TM Address to whitelist transfers
+  const generalTMAddress = (await tickerSecurityTokenInstance.getModulesByName({
+    moduleName: ModuleName.GeneralTransferManager,
   }))[0];
+  const generalTM = await polymathAPI.moduleFactory.getModuleInstance({
+    name: ModuleName.GeneralTransferManager,
+    address: generalTMAddress,
+  });
+
+  // Add owner address in the whitelist to allow issue tokens
+  await generalTM.modifyKYCData({
+    investor: myAddress,
+    canSendAfter: new Date(),
+    canReceiveAfter: new Date(),
+    expiryTime: new Date(2020, 0),
+    txData: {
+      from: await polymathAPI.getAccount(),
+    },
+  });
+
+  // Mint yourself some tokens and make some transfers
+  await tickerSecurityTokenInstance.issue({
+    investor: myAddress,
+    value: new BigNumber(50),
+    data: '0x00',
+  });
 
-  await polymathAPI.polyToken.transfer({ to: erc20Dividend, value: new BigNumber(5) });
   await polymathAPI.polyToken.approve({
-    spender: erc20Dividend,
+    spender: erc20DividendAddress,
     value: new BigNumber(4),
   });
 
+  const randomInvestors = [
+    '0x1111111111111111111111111111111111111111',
+    '0x2222222222222222222222222222222222222222',
+    '0x3333333333333333333333333333333333333333',
+  ];
+
+  // Add beneficiaries address to whitelist
+  await generalTM.modifyKYCDataMulti({
+    investors: randomInvestors,
+    canSendAfter: [new Date(), new Date(), new Date()],
+    canReceiveAfter: [new Date(), new Date(), new Date()],
+    expiryTime: [new Date(2020, 0), new Date(2020, 0), new Date(2020, 0)],
+  });
+
+  await tickerSecurityTokenInstance.transfer({ to: randomInvestors[0], value: new BigNumber(10) });
+  await tickerSecurityTokenInstance.transfer({ to: randomInvestors[1], value: new BigNumber(20) });
+  await tickerSecurityTokenInstance.transfer({ to: randomInvestors[2], value: new BigNumber(20) });
+
   // Create Dividends
   await erc20DividendCheckpoint.createDividendWithExclusions({
-    name: 'MyDividend2',
+    name: 'MyDividend1',
     amount: new BigNumber(1),
     token: await polymathAPI.polyToken.address(),
     expiry: new Date(2035, 2),
     maturity: new Date(2018, 1),
-    excluded: ['0x1111111111111111111111111111111111111111', '0x2222222222222222222222222222222222222222'],
+    excluded: [randomInvestors[1], randomInvestors[2]],
   });
 
+  // Create a checkpoint
+  await tickerSecurityTokenInstance.createCheckpoint({});
```

Comment:
The fact that you create a checkpoint (which would be checkpointId 2 since the call to `createDividendWithExclusions` created checkpointId 1 behind the scenes) and then don't use it might be confusing to the users. I would add comments explaining that `createDividendWithExclusions` creates a checkpoint behind the scenes, and so the newly created checkpoint would be nº 2. I would also use 2 as the checkpointId (at least for some of the dividend distributions)

Reply:
Alright I think I understand. I have updated some comments and used different checkpoint ids, please check

Reply:
Checkpoints are 1-based, not 0-based. So the next two calls that use ids should use 2 and 3, respectively